Mage.Sets/src/mage/cards/a/AtomwheelAcrobats.java

```diff
@@ -0,0 +1,156 @@
+
+        // Whenever you roll a 1 or 2, put that many +1/+1 counters on Atomwheel Acrobats.
+        this.addAbility(new AtomwheelAcrobatsTriggeredAbility());
+
+        // {2}{G}: Roll a six-sided die.
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new AtomwheelAcrobatsDieRollEffect(), new ManaCostsImpl<>("{2}{G}"));
+        this.addAbility(ability);
+    }
+
+    private AtomwheelAcrobats(final AtomwheelAcrobats card) {
+        super(card);
+    }
+
+    @Override
+    public AtomwheelAcrobats copy() {
+        return new AtomwheelAcrobats(this);
+    }
+}
+
+// Based on Mr. House, President and CEO
+class AtomwheelAcrobatsTriggeredAbility extends TriggeredAbilityImpl {
+
+    public AtomwheelAcrobatsTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new AtomwheelAcrobatsCountersEffect(), false);
+    }
+
+    private AtomwheelAcrobatsTriggeredAbility(final AtomwheelAcrobatsTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public AtomwheelAcrobatsTriggeredAbility copy() {
+        return new AtomwheelAcrobatsTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DIE_ROLLED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        DieRolledEvent drEvent = (DieRolledEvent) event;
+        if (this.isControlledBy(event.getTargetId()) && drEvent.getRollDieType() == RollDieType.NUMERICAL) {
+            int result = drEvent.getResult();
+            if (result == 1 || result == 2) {
+                this.getEffects().setValue("rolled", result);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public String getRule() {
+        return "Whenever you roll a 1 or 2, put that many +1/+1 counters on Atomwheel Acrobats.";
+    }
+}
+
+class AtomwheelAcrobatsCountersEffect extends OneShotEffect {
+
+    AtomwheelAcrobatsCountersEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private AtomwheelAcrobatsCountersEffect(final AtomwheelAcrobatsCountersEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AtomwheelAcrobatsCountersEffect copy() {
+        return new AtomwheelAcrobatsCountersEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (getValue("rolled") == null) {
+            return false;
+        }
+        int amount = (Integer) getValue("rolled");
+        Effect effect = null;
+        if (amount == 1) {
+            effect = new AddCountersSourceEffect(CounterType.P1P1.createInstance(1));
+        }
+        if (amount == 2) {
+            effect = new AddCountersSourceEffect(CounterType.P1P1.createInstance(2));
+        }
+        if (effect != null) {
+            effect.apply(game, source);
+            return true;
+        }
+        return amount >= 1;
+    }
```

Comment:
it's a lot easier here to just get the source permanent with `source.getSourcePermanentIfItStillExists(game)` and add the counters directly using the `amount` value